test/protocol.spec.ts

```diff
@@ -27,11 +27,12 @@ describe('protocol-contract', async () => {
     });
 
     r = await d.protocol.createDeploymentSubset(DEPLOYMENT_SUBSET_CANARY, 2);
+    currTime = await getTopBlockTimestamp(d);
     expect(r).to.have.a.protocolChangedEvent({
       deploymentSubset: DEPLOYMENT_SUBSET_CANARY,
       currentVersion: bn(2),
       nextVersion: bn(2),
-      fromTimestamp: bn(r.blockNumber)
+      fromTimestamp: bn(currTime) // assuming ganache will not mine since last tx
```

Comment:
You can take the exact timestamp from the transaciton receipt (`r`). I think there should be somewhere a function named `txTimestamp`.